src/sentry/api/endpoints/organization_member_invite/reinvite.py

```diff
@@ -0,0 +1,150 @@
+            organization=organization,
+            email=invited_member.email,
+            user=request.user,
+            auth=request.auth,
+        ):
+            metrics.incr(
+                "member-invite.attempt",
+                instance="rate_limited",
+                skip_internal=True,
+                sample_rate=1.0,
+            )
+            return Response({"detail": ERR_RATE_LIMITED}, status=429)
+
+        if trigger_regenerate_token:
+            if request.access.has_scope("member:admin"):
+                with transaction.atomic(router.db_for_write(OrganizationMemberInvite)):
+                    invited_member.regenerate_token()
+                    invited_member.save()
+            else:
+                return Response({"detail": ERR_INSUFFICIENT_SCOPE}, status=400)
+
+        if invited_member.token_expired:
+            return Response({"detail": ERR_EXPIRED}, status=400)
+
+        invited_member.send_invite_email()
+
+        self.create_audit_entry(
+            request=request,
+            organization=organization,
+            target_object=invited_member.id,
+            event=audit_log.get_event_id("MEMBER_REINVITE"),
+            data=invited_member.get_audit_log_data(),
+        )
+        return Response(serialize(invited_member, request.user), status=200)
+
+    def put(
+        self,
+        request: Request,
+        organization: Organization,
+        invited_member: OrganizationMemberInvite,
+    ) -> Response:
+        """
+        Resend a member invite to an Organization
+        ````````````````````````````````````````
+
+        Resend an invite to an organization.
+
+        :pparam string organization_id_or_slug: the id or slug of the organization the member will belong to
+        :param string member_id: the member ID
+        :param boolean trigger_regenerate_token: whether to regenerate the member's invite token
+        :auth: required
+        """
+        if not features.has(
+            "organizations:new-organization-member-invite", organization, actor=request.user
+        ):
+            return Response({"detail": MISSING_FEATURE_MESSAGE}, status=403)
+
+        validator = OrganizationMemberReinviteRequestValidator(
+            data=request.data,
+        )
+        if not validator.is_valid():
+            return Response(validator.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        result = validator.validated_data
+
+        actor_is_member = not request.access.has_scope("member:admin") and (
+            request.access.has_scope("member:invite")
+        )
+        # Members can only resend invites that they sent
+        is_invite_from_actor = invited_member.inviter_id == request.user.id
+        org_allows_invites_from_members = not organization.flags.disable_member_invite
+
+        if actor_is_member:
+            if not org_allows_invites_from_members or not is_invite_from_actor:
+                raise PermissionDenied
```

Comment:
nit: collapse into a single if?
```suggestion
        if actor_is_member and (not org_allows_invites_from_members or not is_invite_from_actor):
                raise PermissionDenied
```